src/ImageSharp.Web/Caching/IImageCache.cs

```diff
@@ -23,29 +23,30 @@ public interface IImageCache
         /// <summary>
         /// Gets the value associated with the specified key.
         /// </summary>
-        /// <param name="key">The cache key</param>
-        /// <returns>The <see cref="Task{IByteBuffer}"/></returns>
-        Task<IByteBuffer> GetAsync(string key);
+        /// <param name="key">The cache key.</param>
+        /// <returns>The <see cref="Task{IImageCache}"/>.</returns>
+        Task<ICachedImage> GetAsync(string key);
```

Comment:
I wonder if we need to keep the getting operation async. 

Maybe we should make the `Open*****()` operations async intead, but only if retrieving a Redis or Azure blob store stream is an asynchronous operation.

Reply:
I did think about this actually. It depends basically on when users are likely to get the `LastWriteTimeUtc` property. I think it should always be  lazily in the `ICachedImage` or whatever we call it.